server/data/prisonApi.ts

```diff
@@ -8,14 +8,60 @@ interface CaseLoad {
   type: string
 }
 
+// Possibly not an exhaustive list
+type LocationType =
+  | 'AREA'
+  | 'ADJU'
+  | 'ASSO'
+  | 'SPOR'
+  | 'LOCA'
+  | 'STOR'
+  | 'CLAS'
+  | 'WORK'
+  | 'EXER'
+  | 'IGRO'
+  | 'RESI'
+  | 'ROOM'
+  | 'WING'
+  | 'MEDI'
+  | 'CELL'
+  | 'INTE'
+  | 'BOX'
+  | 'RTU'
+  | 'VIDE'
+  | 'TRAI'
+  | 'LAND'
+  | 'SPUR'
+  | 'HOLD'
+
+interface Location {
+  locationId: number
+  locationType: LocationType
+  description: string
+  agencyId: string
+  currentOccupancy: number
+  operationalCapacity?: number
+  locationPrefix: string
+  userDescription?: string
+}
+
 class PrisonApi extends RestClient {
   constructor(token: string) {
     super('HMPPS Prison API', config.apis.hmppsPrisonApi, token)
   }
 
   async getUserCaseLoads(): Promise<Array<CaseLoad>> {
-    return this.get({ path: '/users/me/caseLoads' }) as Promise<Array<CaseLoad>>
+    return this.get({ path: '/api/users/me/caseLoads' }) as Promise<Array<CaseLoad>>
+  }
+
+  async getAgencyLocations(agencyId: string): Promise<Array<Location>> {
+    const locations = (await this.get({ path: `/api/agencies/${agencyId}/locations` })) as Array<Location>
+
+    // Only return occupied wings
+    return locations.filter(location => {
+      return location.locationType === 'WING' && location.currentOccupancy > 0
```

Comment:
Not sure if we should add a new endpoint to just get wings or you can simply call `{{prison_api}}/api/users/me/locations` which will give you the current users wing locations for the prison they have selected

Reply:
Not sure about the occupancy check - is that a requirement?

Reply:
The occupancy check wasn't a requirement but when I was playing with the API we saw that some wings were repeated/similar. So this is to avoid showing empty wings. At this stage not much point in allowing a user to select a wing which has no data IMHO.